doubly_linked_list.py

```
@@ -0,0 +1,69 @@
+# http://ls.pwd.io/2014/08/singly-and-doubly-linked-lists-in-python/
+
+
+class Node(object):
+    def __init__(self, data):
+        self.next = None
+        self.prev = None
+        self.data = data
+
+
+class DoublyList(object):
+    def __init__(self):
+        self.head = None
+        self.tail = None
+
+    def insert(self, data):
+        """Insert data to at head"""
+        node = Node(data)
+        if self.head is None:
+            self.head = self.tail = node
+        elif self.head is not None:
+            node.prev = self.head
+            self.head.prev = node
+            self.head = node
+
+    def append(self, data):
+        """Insert data to at head"""
+        node = Node(data)
+        if self.head is None:
+            self.head = self.tail = node
+        elif self.head is not None:
+            node.next = self.tail
+            self.tail.next = node
+            self.tail = node
+
+    def pop(self):
+        """Remove and return head value"""
```

This should throw an error on an empty list so the user knows that the function did not work. Same with shift()